src/services/fileStorage/proxy-service.js

```diff
@@ -151,29 +152,41 @@ const fileStorageService = {
 	 */
 	async create(data, params) {
 		const {
-			payload: { userId, fileStorageType },
+			payload: { userId: creatorId, fileStorageType },
 		} = params;
 		const { owner, parent, studentCanEdit, permissions: sendPermissions = [] } = data;
 
 		const refOwnerModel = await getRefOwnerModel(owner);
 
-		const permissions = await createDefaultPermissions(userId, refOwnerModel, {
+		const permissions = await createDefaultPermissions(creatorId, refOwnerModel, {
 			studentCanEdit,
 			sendPermissions,
 			owner,
 		}).catch((err) => {
 			throw new GeneralError('Can not create default Permissions', err);
 		});
 
+		const strategy = createCorrectStrategy(fileStorageType);
+
+		const fileOwner = owner || creatorId;
+
+		const creator = await userModel.findById(creatorId).lean().exec();
+		const { schoolId } = creator;
+		const bucket = strategy.getBucket(schoolId);
+		const school = await schoolModel.findById(schoolId).lean().exec();
+		const storageProviderId = school.storageProvider;
```

Comment:
can the school's bucket or storage provider ever change? How to handle the files in that case?

Reply:
The functionality is provided via SHD, but it is currently not really supported. It can only be used, if there are no files saved in the storage provider yet.